### PR TITLE
Improvements to contract factory

### PR DESCRIPTION
**Pull request by rodrigo-pino**
- The implicit conversion pass is no longer a requirement
- Salt (bytes32) is now converted to a bytes30, now it no longer errors
  for values bigger than a felt
- calldata expected location is set only for reference types

### CODE REVIEW

**Comment by GuyMcComb** [comment]
Running Semantic Tests

File: tests/behaviour/expectations/behaviour.ts [hunk 1]
```diff
@@ -3257,8 +3257,9 @@ export const expectations = flatten(
           ]),
           File.Simple('salted', [
             new Expect('Create multiple salted contracts', [
-              ['createUint256Contract', ['0', '1', '0', '2'], [], '0'],
-              ['createUint256Contract', ['0', '2', '0', '5'], [], '0'],
+              ['createAndStoreUint256Contract', ['0', '1', '0', '2'], [], '0'],
+              ['createAndStoreUint256Contract', ['0', '2', '0', '5'], [], '0'],
+              ['createUint256Contract', ['0', `${2n ** 128n - 1n}`, '0', '7'], [], '0'],
             ]),
```
[the remark this answers is not in the document]
Comment (rodrigo-pino) [reply tests/behaviour/expectations/behaviour.ts:3263]: The test `createUint256Contract` receives 2**256 -1

[the remark this answers is not in the document]
Comment (GuyMcComb) [reply tests/behaviour/expectations/behaviour.ts:3263]: Oh sorry I see now. But receives (2**128-1)*2**128 but the effect is the same.

**Comment by GuyMcComb** [comment]
Sem Tests pass

**Review by GuyMcComb — APPROVED** [review]
Good work

**Pull request merged**